src/Model/ChangeStreamIterator.php

```diff
@@ -122,12 +125,13 @@ final public function commandSucceeded(CommandSucceededEvent $event): void
 
     /**
      * @see https://php.net/iteratoriterator.current
-     * @return mixed
+     * @return array|object|null
+     * @psalm-return TValue|null
      */
     #[ReturnTypeWillChange]
     public function current()
     {
-        return $this->isValid ? parent::current() : null;
+        return $this->valid() ? parent::current() : null;
```

Comment:
Was this necessary to use the `@psalm-assert-if-true` annotation on `valid()`? Otherwise, I see no reason to add a method call.

Reply:
That is correct. I'll do some checking to see if `@psalm-assert-if-true` also works on variables and if psalm picks up on that to avoid the method call.

Reply:
Did some checking on the Playground, and unfortunately it doesn't work with variables: https://psalm.dev/r/bf4920bc0b